src/librbd/CMakeLists.txt

```diff
@@ -153,6 +152,12 @@ set(librbd_internal_srcs
   watcher/RewatchRequest.cc
   ${CMAKE_SOURCE_DIR}/src/common/ContextCompletion.cc)
 
+if (NOT WIN32)
+  list(
+    APPEND librbd_internal_srcs
+    cache/ParentCacheObjectDispatch.cc)
```

Comment:
I am just wondering, do we really need to disable this on Windows? I suppose there is no problem with compiling `cache/ParentCacheObjectDispatch.cc` on Windows? Then why not leave it here? It seems the only thing you need is to disable building `immutable_object_cache` tool and make sure that that "rbd_parent_cache_enabled" option can only be set to `false` on Windows. For the last I suggest to add a validator for "rbd_parent_cache_enabled" option in `common/options.cc` [1]. See, e.g. the validator for "rbd_default_pool" option [2] as an example. This way we would not need any "ifdefs" in librbd.

librbd/image/OpenRequest.cc 

[1] https://github.com/ceph/ceph/blob/master/src/common/options.cc#L7078
[2] https://github.com/ceph/ceph/blob/master/src/common/options.cc#L6974

Reply:
Ah, it will not link due to missing ceph_immutable_object_cache_lib.

Then probably provide the cache client libriry (something like  ceph_immutable_object_cache_client_lib) and have Windows ifdefs there, compiling dumb functions for Windows until the socket support is added? This looks cleaner than including `librbd/cache/ParentCacheObjectDispatch.cc` in `librbd/image/OpenRequest.cc`. These changes in `OpenRequest.cc` is what I actually do not like much. Ideally it should not have any Windows specific code.

Reply:
Thanks for the suggestion. Completely disabling this on Windows seems safer and much easier than stubbing the object cache API. I can do that if really necessary, although it may require significantly more changes than the current approach.

Reply:
My main concern is including `librbd/cache/ParentCacheObjectDispatch.cc` in `librbd/image/OpenRequest.cc`. This looks ugly to me. Also, I would prefer avoiding ifdefs in files that do not have platform specific code (like OpenRequest.cc).

But saying this I don't have strong objection against the current code, it looks good enough to me to be merged as it is, and can be improved later.

Reply:
Good news, we can let boost use unix sockets on Windows by changing just a few lines: https://github.com/huangqinjin/asio/commit/d27a8ad1870da07f93331639ffcea33ce951f648

I'll apply this as part of the Windows build and drop those platform checks.